tests/kafkatest/tests/verifiable_consumer_test.py

```diff
@@ -72,16 +86,59 @@ def await_produced_messages(self, producer, min_messages=1000, timeout_sec=10):
                    err_msg="Timeout awaiting messages to be produced and acked")
 
     def await_consumed_messages(self, consumer, min_messages=1):
+        timeout_sec = self.consumption_timeout_sec
         current_total = consumer.total_consumed()
-        wait_until(lambda: consumer.total_consumed() >= current_total + min_messages,
-                   timeout_sec=self.consumption_timeout_sec,
-                   err_msg="Timed out waiting for consumption")
+        expected = current_total + min_messages
+
+        def _condition():
+            return consumer.total_consumed() >= expected
+
+        def _err_msg():
+            actual = consumer.total_consumed()
+            return "%d messages received within the timeout of %d seconds, expected %d" % (actual, timeout_sec, expected)
+
+        wait_until(lambda: _condition(), timeout_sec=timeout_sec, err_msg=_err_msg())
+
+    def await_members_stopped(self, consumer, num_consumers, timeout_sec):
+        self._await_members_in_state(consumer, num_consumers, "stopped", [ConsumerState.Dead], timeout_sec)
 
     def await_members(self, consumer, num_consumers):
         # Wait until all members have joined the group
-        wait_until(lambda: len(consumer.joined_nodes()) == num_consumers,
-                   timeout_sec=self.session_timeout_sec*2,
-                   err_msg="Consumers failed to join in a reasonable amount of time")
-        
+        states = [ConsumerState.Joined]
+
+        if consumer_group.is_consumer_group_protocol_enabled(consumer.group_protocol):
+            states.extend([ConsumerState.Started, ConsumerState.Rebalancing])
```

Comment:
Why do we have to do this? Are we still checking the same thing if we consider "started" as "joined"?